### DISCUSSION TITLE
powersave option tasmota-sensors.bin.gz 13.1.0 (HC-SR04 on ESP8266 battery powered)

### DISCUSSION THREAD

**Discussion by grericht** [arendst/Tasmota#19610] [Q&A]
Hello, I have connected an HC-SR04 to an ESP8266, and it's working well. I'm using it to report the water level in a rain barrel to Home Assistant via MQTT, and that's functioning as expected. The telemetry period is set to 300 seconds, which is sufficient for my needs. Is it possible to put the sensor into deep sleep mode during the intervals? Currently, it's continuously consuming 0.35W. It should be possible to wake the sensor briefly every 300 seconds and conserve power in between, allowing me to operate the sensor on battery power.

edit: The command "powersave 1" in the console returns "unknown."

**Comment by sfromis** [comment]
https://tasmota.github.io/docs/DeepSleep/ 
However, in general Tasmota is not designed for low power operation.

When it comes to the sensor (which is not a great choice for low power operation) managing power supplied to that is a hardware issue outside the control of Tasmota. Sure, at the circuit level, you could do "something", possibly including transistors controlled via a gpio pin configured as `Output Hi` active while the chip is out of deep sleep and Tasmota boots again.
